From 3e966b70fba397e7a494a94e774cf50a62239da6 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 18:58:55 -0400
Subject: [fs] btrfs: fix extent_map block_len after merging

Message-id: <1407179109-13714-49-git-send-email-sandeen@redhat.com>
Patchwork-id: 86992
O-Subject: [RHEL7.1 PATCH 048/422] Btrfs: fix extent_map block_len after merging
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Filipe David Borba Manana <fdmanana@gmail.com>

commit d527afe1e5865d25a336f6c4157f1086cfbddc81
Author: Filipe David Borba Manana <fdmanana@gmail.com>
Date:   Sat Nov 30 11:28:35 2013 +0000

    Btrfs: fix extent_map block_len after merging

    When merging an extent_map with its right neighbor, increment
    its block_len with the neighbor's block_len.

    Signed-off-by: Filipe David Borba Manana <fdmanana@gmail.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/extent_map.c b/fs/btrfs/extent_map.c
index b60955d..996ad56 100644
--- a/fs/btrfs/extent_map.c
+++ b/fs/btrfs/extent_map.c
@@ -255,7 +255,7 @@ static void try_merge_map(struct extent_map_tree *tree, struct extent_map *em)
   merge = rb_entry(rb, struct extent_map, rb_node);
  if (rb && mergable_maps(em, merge)) {
   em->len += merge->len;
-  em->block_len += merge->len;
+  em->block_len += merge->block_len;
   rb_erase(&merge->rb_node, &tree->map);
   merge->in_tree = 0;
   em->mod_len = (merge->mod_start + merge->mod_len) - em->mod_start;
-- 
1.7.1